todocli/datetime_parser.py

```diff
@@ -9,31 +9,118 @@ def parse_hour_minute(input_str):
-        split_str = datetime_str.split(" ")
-        day, month = parse_day_month(split_str[0])
-        hour, minute = parse_hour_minute(split_str[1])
-        return datetime.now().replace(day=day, month=month, hour=hour, minute=minute)
+    try:
+        if re.match(r"([0-9]{1,2}h)", datetime_str, re.IGNORECASE):
+            """ e.g. 1h / 12h """
+            return datetime.now() + timedelta(hours=int(datetime_str[:-1]))
+
+        if datetime_str == "morning":
+            dt = datetime.now()
+            return add_day_if_past(
+                dt.replace(hour=7, minute=0, second=0, microsecond=0)
+            )
+
+        if datetime_str == "tomorrow":
+            dt = datetime.now()
+            return dt.replace(hour=7, minute=0, second=0, microsecond=0) + timedelta(
+                days=1
+            )
+
+        if datetime_str == "evening":
+            dt = datetime.now()
+            return add_day_if_past(
+                dt.replace(hour=18, minute=0, second=0, microsecond=0)
+            )
+
+        if re.match(r"([0-9]{1,2}:[0-9]{2} (am|pm))", datetime_str, re.IGNORECASE):
+            """ e.g. 5:30 pm """
+            split_str = datetime_str.split(" ")
+            hour, minute = parse_hour_minute(split_str[0])
+
+            if split_str[1] == "am":
+                if hour == 12:
+                    hour = 0
+            else:
+                hour = hour + 12
+
+            return add_day_if_past(
+                datetime.now().replace(
+                    hour=hour, minute=minute, second=0, microsecond=0
+                )
+            )
+
+        if re.match(r"([0-9]{1,2}:[0-9]{2})", datetime_str, re.IGNORECASE):
+            """ e.g. 17:00 """
+            hour, minute = parse_hour_minute(datetime_str)
+            return add_day_if_past(
+                datetime.now().replace(
+                    hour=hour, minute=minute, second=0, microsecond=0
+                )
+            )
+
+        if re.match(
+            r"([0-9]{1,2}\.[0-9]{1,2}\. [0-9]{1,2}:[0-9]{2})",
+            datetime_str,
+            re.IGNORECASE,
+        ):
+            """ e.g. 17.01. 17:00 """
+            split_str = datetime_str.split(" ")
+            day, month = parse_day_month_DD_MM(split_str[0])
+            hour, minute = parse_hour_minute(split_str[1])
+            return datetime.now().replace(
+                day=day, month=month, hour=hour, minute=minute, second=0, microsecond=0
+            )
+
+        if re.match(
+            r"([0-9]{1,2}/[0-9]{1,2} [0-9]{1,2}:[0-9]{2} (am|pm))",
+            datetime_str,
+            re.IGNORECASE,
+        ):
+            """ e.g. 17/01 5:00 pm """
+            split_str = datetime_str.split(" ")
+            day, month = parse_day_month_MM_DD(split_str[0])
+            hour, minute = parse_hour_minute(split_str[1])
+            return datetime.now().replace(
+                day=day, month=month, hour=hour, minute=minute, second=0, microsecond=0
+            )
```

Comment:
Let's remove this functionality (or comment it out) for now since 'month-day' format has been removed from the documentation. Also, I think the example should be `01/17`?

Reply:
Done